File: PHP/Challenge/September/02_Find_the_Student_that_Will_Replace_the_Chalk.php
<?php
// https://leetcode.com/problems/find-the-student-that-will-replace-the-chalk/description/?envType=daily-question&envId=2024-09-02

function chalkReplacer($chalk, $k)
{
    $total_sum = array_sum($chalk);
    $k %= $total_sum;
    for ($i = 0; $i < count($chalk); $i++) {
        if ($k < $chalk[$i]) {
            return $i;
        }
        $k -= $chalk[$i];
    }
}

//$chalk = [5, 1, 5];
//$k = 22;
$chalk = [3, 4, 1, 2];
$k = 25;
echo chalkReplacer($chalk, $k);

File: Python/Challenge/September/02_Find_the_Student_that_Will_Replace_the_Chalk.py
# https://leetcode.com/problems/find-the-student-that-will-replace-the-chalk/description/?envType=daily-question&envId=2024-09-02
from typing import List


def chalkReplacer(chalk: List[int], k: int) -> int:
    total_chalk = sum(chalk)
    # k = k % total_chalk
    k %= total_chalk
    for i, c in enumerate(chalk):
        if k < c:
            return i
        k -= c


# chalk = [5, 1, 5]
# k = 22

chalk = [3, 4, 1, 2]
k = 25
print(chalkReplacer(chalk, k))
